### DISCUSSION TITLE
Feature Request: Include 'SameSite=none' Attribute in Set Cookies

### DISCUSSION THREAD

**Discussion by jlizancandela** [xavierbriole/react-cookienotice#199] [Ideas]
Hello! First, I want to express my gratitude for all the work you've put into this React component and for sharing it with the community.

I'd like to suggest a small enhancement for future releases. Since browsers are implementing changes to cookie policies, it would be beneficial to include the "SameSite=none" attribute in cookies set by this component. This will help ensure its compatibility with upcoming changes and avoid potential security issues in the future.

Thank you again for your contribution and for considering this suggestion!

Best regards,
jlizancandela

**Comment by xavierbriole** [comment]
Hello Jorge,

Thanks a lot! It's always a pleasure to share some code with the OpenSource community.

This is a great idea, I wonder if we can go even further by giving the ability to set some others flags like `HttpOnly`, `Secure` and for sure, the one you suggested, `SameSite` ? Then in `SameSite` you can set it to `Strict`, `Lax` or `None`.

What do you think ?

Thank you for your suggestion, that help the react-cookienotice package to be better because it's very quiet since a moment here…

— Xavier

**Reply by jlizancandela** [reply]
Hello Xavier,

Thank you for your response and for considering my suggestion! I completely agree with you. Expanding the configuration options to include attributes like HttpOnly, Secure, and different options for SameSite would be really beneficial. This would allow users to adjust the security level according to their specific needs. I think it would be an excellent addition to the react-cookienotice package.

Thanks again for your consideration and for your work on this project.Best regards, jlizancandela.

**Reply by xavierbriole** [reply]
Hello,

The feature has been released into the [v7.0.0-beta.0](https://github.com/xavierbriole/react-cookienotice/releases/tag/v7.0.0-beta.0) version.

I'll test it asap.

See you!
